kong/runloop/handler.lua

```diff
@@ -1513,7 +1513,10 @@ return {
       -- Report HTTP status for health checks
       local balancer_data = ctx.balancer_data
       if balancer_data and balancer_data.balancer_handle then
-        local status = ngx.status
+        -- https://github.com/nginx/nginx/blob/2485681308bd8d3108da31546cb91bb97813a3fb/src/http/ngx_http_upstream.c#L5656
+        -- because of the way of Nginx do the upstream_status variable, it may be
+        -- a string or a number, so we need to parse it to get the status
+        local status = tonumber(sub(var.upstream_status or "", -3)) or ngx.status
```

Comment:
Should we consider moving this into the PDK or some other helper function?
I can see other modules, especially around metrics and logging that would like to use this capability.

Reply:
Good idea. Perhaps we should do it in another PR and not this.

Reply:
Thanks @hbagdi , I will open another PR to do this.

Reply:
a link to possible values of `var.upstream_status` (like http://nginx.org/en/docs/http/ngx_http_upstream_module.html#var_upstream_addr) instead of pointing to code would be more straightforward for future us